src/app/market/sell/add-item/add-item.component.ts

```diff
@@ -315,6 +315,14 @@ export class AddItemComponent implements OnInit, OnDestroy {
     return this.itemFormGroup.valid || this.snackbar.open('Invalid Listing');
   }
 
+  inputValidate(event: any) {
```

Comment:
If using the same method handler to cater for paste events, an example updated function might look something as follows:

```
inputValidate(event: any) {
    // Special character validation
    const keys = ['-', 'e', 'E', '+'];
    const pasted = String(event.clipboardData ? event.clipboardData.getData('text') : '' );
    if (keys.includes(event.key) || pasted.split('').find((c) =>  keys.includes(c))) {
      return false;
    }
  }
```

I would also suggest renaming this function to indicate that the validation is for numerical fields.

Reply:
@zaSmilingIdiot for me on mac clipboardData doesn't return any data not sure why :(